python/lsst/ts/standardscripts/scheduler/base_set_desired_state.py

```diff
@@ -0,0 +1,214 @@
+# This file is part of ts_standardscripts
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+
+__all__ = ["BaseSetDesiredState"]
+
+import abc
+import types
+import typing
+import asyncio
+
+from lsst.ts import salobj
+
+
+class BaseSetDesiredState(salobj.BaseScript):
+    """A base script that implements setting the desired state for the
+    Scheduler.
+
+    Parameters
+    ----------
+    index : `int`
+        Index of Script SAL component.
+    descr : `str`
+        Script description.
+    scheduler_index : `int`
+        Index of the Scheduler to enable.
+    """
+
+    def __init__(self, index: int, descr: str, scheduler_index: int) -> None:
+        super().__init__(index=index, descr=descr)
+
+        self.scheduler_remote = salobj.Remote(
+            domain=self.domain,
+            name="Scheduler",
+            index=scheduler_index,
+            include=["summaryState", "heartbeat"],
+        )
+
+        self.timeout_start = 30.0
+
+        self._tasks_to_handle_summary_state = (
```

Comment:
Please consider reserving "task" for instances of asyncio.Task or Future.

If you keep this code, please consider a name such as `self._state_transition_methods_to_try`